Rematch/Addon.lua

```diff
@@ -165,30 +242,99 @@ function RematchPlugin:OnTooltipFormatting(tip, key)
 
+function RematchPlugin:OnExportImpl(key)
+    local rematchVersion = ns.Version:Current('Rematch')
+
+    if rematchVersion >= ns.Version:New(5, 0, 0, 0) then
+        return Rematch.teamStrings:ExportTeam(key)
+    else
+        Rematch:SetSideline(key, self.savedRematchTeams[key])
+        return Rematch:ConvertSidelineToString()
+    end
+end
+
 function RematchPlugin:OnExport(key)
     local rematchVersion = ns.Version:Current('Rematch')
 
     if not rematchVersion then
         return
     end
 
-    if rematchVersion < ns.Version:New(5, 0, 0, 0) then
-        if self.savedRematchTeams[key] then
-            Rematch:SetSideline(key, self.savedRematchTeams[key])
-            return Rematch:ConvertSidelineToString()
-        end
+    if not self.savedRematchTeams[key] then
+        return
     end
+
+    return self:OnExportImpl(key)
 end
 
-function RematchPlugin:OnImport(data)
+local importTeamIdCounter = 0
+local importTeamIdMapping = {}
+
+function RematchPlugin:OnImport(script, extra)
     local rematchVersion = ns.Version:Current('Rematch')
 
     if not rematchVersion then
         return
     end
 
-    if rematchVersion < ns.Version:New(5, 0, 0, 0) then
+    if rematchVersion >= ns.Version:New(5, 0, 0, 0) then
+        local temporaryTeamId = 'temporary-team-id-' .. importTeamIdCounter
+        importTeamIdCounter = importTeamIdCounter + 1
+        self:MoveScript(script:GetKey(), temporaryTeamId)
+        importTeamIdMapping[extra] = temporaryTeamId
+
+        Rematch.dialog:ShowDialog('ImportTeams')
+        Rematch.dialog.Canvas.MultiLineEditBox:SetText(extra)
+    else
         Rematch:ShowImportDialog()
-        RematchDialog.MultiLine.EditBox:SetText(data)
+        RematchDialog.MultiLine.EditBox:SetText(extra)
+    end
+end
+
+function RematchPlugin:OnImportContinuation(addedTeamID)
+    local extra = self:OnExportImpl(addedTeamID)
+    local temporaryTeamId = importTeamIdMapping[extra]
+
+    if not temporaryTeamId then
+        return
+    end
+
+    importTeamIdMapping[extra] = nil
+
+    self:MoveScript(temporaryTeamId, addedTeamID)
+end
+
+function RematchPlugin.ResetEverything()
+    local scriptsDB = PetBattleScripts.db.global.scripts
+
+    scriptsDB.Rematch = CopyTable(scriptsDB.Rematch4)
+    wipe(scriptsDB.Rematch4)
+end
+
+function RematchPlugin:UpdateDBRematch4To5(convertedTeams)
+    -- Backup old scripts.
+    local scriptsDB = PetBattleScripts.db.global.scripts
+    scriptsDB.Rematch4 = CopyTable(scriptsDB.Rematch)
+
+    -- First we need a cache list of all of our scripts, so we can modify our scripts
+    -- database without messing up the loops.
+    local scriptList = {}
+    for key, script in self:IterateScripts() do
+        scriptList[key] = script
+    end
+
+    -- Second we can migrate scripts.
+    for oldTeamID, script in pairs(scriptList) do
+        local newTeamID = convertedTeams[oldTeamID]
+        if newTeamID then
+            self:MoveScript(oldTeamID, newTeamID)
+        end
+    end
+
```

Comment:
Do you want to update the script name (`overwriteName`) durnin migration, or keep them?